packages/docusaurus/src/webpack/plugins/WaitPlugin.ts

```diff
@@ -23,21 +21,34 @@ export default class WaitPlugin {
 
   apply(compiler: Compiler): void {
     // Before finishing the compilation step
-    compiler.hooks.make.tapAsync('WaitPlugin', (compilation, callback) => {
-      // To prevent 'waitFile' error on waiting non-existing directory
-      fs.ensureDir(path.dirname(this.filepath), {}, () => {
-        // Wait until file exist
-        waitOn({
-          resources: [this.filepath],
-          interval: 300,
-        })
-          .then(() => {
-            callback();
-          })
-          .catch((error: Error) => {
-            console.warn(`WaitPlugin error: ${error}`);
-          });
-      });
+    compiler.hooks.make.tapPromise('WaitPlugin', () => waitOn(this.filepath));
+  }
+}
+
+async function waitOn(filepath: string): Promise<void> {
```

Comment:
If you copied the code or adapted the code, add a comment linking to the original source both for attribution and for backtracing purposes